src/Components/test/testassets/Ignitor/BlazorClient.cs

```diff
@@ -75,55 +163,92 @@ public async Task<bool> ConnectAsync(Uri uri, bool prerendered)
             if (prerendered)
             {
                 CircuitId = await GetPrerenderedCircuitIdAsync(uri);
-                return await HubConnection.InvokeAsync<bool>("ConnectCircuit", CircuitId);
+                var result = false;
+                await ExpectRenderBatch(async () => result = await HubConnection.InvokeAsync<bool>("ConnectCircuit", CircuitId));
+                return result;
             }
             else
             {
-                CircuitId = await HubConnection.InvokeAsync<string>("StartCircuit", uri, uri);
+                await ExpectRenderBatch(
+                    async () => CircuitId = await HubConnection.InvokeAsync<string>("StartCircuit", new Uri(uri.GetLeftPart(UriPartial.Authority)), uri),
+                    TimeSpan.FromSeconds(10));
                 return CircuitId != null;
             }
+        }
 
-            void OnBeginInvokeJS(int asyncHandle, string identifier, string argsJson)
+        private void OnBeginInvokeJS(int asyncHandle, string identifier, string argsJson)
+        {
+            try
             {
                 JSInterop?.Invoke(asyncHandle, identifier, argsJson);
+
+                NextJSInteropReceived?.Completion?.TrySetResult(null);
+            }
+            catch (Exception e)
+            {
+                NextJSInteropReceived?.Completion?.TrySetResult(e);
             }
+        }
 
-            void OnRenderBatch(int browserRendererId, int batchId, byte[] batchData)
+        private void OnRenderBatch(int browserRendererId, int batchId, byte[] batchData)
+        {
+            try
             {
+                RenderBatchReceived?.Invoke(browserRendererId, batchId, batchData);
+
                 var batch = RenderBatchReader.Read(batchData);
-                hive.Update(batch);
+
+                Hive.Update(batch);
 
                 if (ConfirmRenderBatch)
                 {
                     HubConnection.InvokeAsync("OnRenderCompleted", batchId, /* error */ null);
                 }
 
-                RenderBatchReceived?.Invoke(browserRendererId, batchId, batchData);
+                NextBatchReceived?.Completion?.TrySetResult(null);
             }
-
-            void OnError(Error error)
+            catch (Exception e)
             {
-                Console.WriteLine("ERROR: " + error.Stack);
+                NextBatchReceived?.Completion?.TrySetResult(e);
```

Comment:
Is there a use case for this? Do we test cases where we expect rendering to fail.